lib/openhab/dsl.rb

```diff
@@ -970,6 +971,24 @@ def try_parse_time_like(string)
 
       raise exception
     end
+
+    #
+    # Provide access to the script context / variables
+    # see OpenHAB::DSL::Rules::AutomationRule#execute!
+    #
+    # @!visibility private
+    ruby2_keywords def method_missing(method, *args)
+      return super unless args.empty? && !block_given?
+      return super unless (context = Thread.current[:openhab_context]) && context.key?(method)
+
+      logger.trace("DSL#method_missing found context variable: '#{method}'")
+      context[method]
+    end
+
+    # @!visibility private
+    def respond_to_missing?(method, include_private = false)
+      ((context = Thread.current[:openhab_context]) && context.key?(method)) || super
```

Comment:
This would be better done with safe navigation:
```suggestion
      Thread.current[:openhab_context]&.key?(method) || super
```

Reply:
Duh Of course! It was a lazy copy paste ;)